modules/travel_pay/app/services/travel_pay/session.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module TravelPay
+  class Session
+    def initialize(client_number, current_user)
+      @client_number = client_number
+      @user = current_user
+    end
+
+    def get_tokens
+      raise NotImplementedError
```

Comment:
This class `TravelPay::Session` is intended to act as an abstract class in order to provide a set of overridable methods for subclasses to implement. I.e. we will never create a new `TravelPay::Session` instance.

It could be I'm borrowing OOP principles from Java and this is not idiomatic Ruby. Thoughts?

Reply:
To that point, and this is more ignorance on my part, but is [this line](https://github.com/department-of-veterans-affairs/vets-api/pull/18677/files#diff-7a072c01e3dfe5c61992e2c2c227c0f64bd355c1cac4f048a473a69e5ef0e800R4) an unnecessary abstraction and the client number and user number can just be initialized there? I.e., are there other subclasses that would need to be created from `TravelPay::Session`, or is the existing `TokenService` all we'll need?

Reply:
The crux! Currently there is no need for a subclass, but I'm afraid for the future when and if the auth mechanism changes on the API-side. I'm going for an open-for-extension, closed-for-modification vibe. Good ol' SOLID principles